### PR TITLE
docs: Add Maintaining AUR packages with Renovate user story 

### LINKED ISSUE #21082: Put "Maintaining AUR packages with Renovate" article into User Stories
### Describe the proposed change(s).

After this PR is merged:

- https://github.com/JamieMagee/jamiemagee.co.uk/pull/124

Do:

- [x] Copy/paste source file from Jamie's repository
- [x] Convert to one sentence per line pattern
- [x] Clearly note that Jamie is the author
- [x] Link to the official source
- [x] Clearly note that Jamie is a maintainer, so people know the article is biased towards Renovate bot
- [x] Put the file in the "User Stories" section
- [x] Update the `docs/usage/user-stories/.pages` file to include the new article
- [x] Let `@JamieMagee` review the article
- [x] Get permission from @JamieMagee to include the article in our docs

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Copy/paste @JamieMagee's article about "Maintaining AUR packages with Renovate" to our User Stories section of the docs
- My changes:
  - Follow one sentence per line pattern
  - Copied the TOC hiding CSS from the Swissquote article
  - Link to the original source
  - Tell the readers that Jamie is one of the maintainers of Renovate bot
  - Add the article to the sidebar of the published docs
  - Use relative links for the datasources and managers "homepages"
  - Update number of datasources in Renovate: `over 50 different datasources`
  - Some more small fixes

## Context

Source file: [permalink to the file on the `JamieMagee/jamiemagee.co.uk` repository](https://github.com/JamieMagee/jamiemagee.co.uk/blob/b940859b311cca9e2317ddd0b3630edf2a68ee24/content/blog/maintaining-aur-packages-with-renovate.md).

Closes #21082

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+I think [Renovate](https://github.com/renovatebot/renovate/) can help here.
+
+## Updating versions with Renovate
+
+Renovate is an automated dependency update tool.
+You might have seen it opening pull requests on GitHub and making updates for npm or other package managers, but it’s a lot more powerful than just that.
+
+Renovate has a couple of concepts that I need to explain first: [datasources](../modules/datasource/index.md) and [managers](../modules/manager/index.md).
+Datasources define where to look for new versions of a dependency.
+Renovate comes with over 50 different datasources, but the one that is important for AUR packages is the [`git-tags` datasource](https://docs.renovatebot.com/modules/datasource/#git-tags-datasource).
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:34]: The source article says "over 40 different datasources", but I'm counting 50+ datasources on the datasources homepage in the docs. Please check my counting. 😄

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+# Maintaining AUR packages with Renovate
+
+> This article was written by [Jamie Magee](https://github.com/JamieMagee) and originally published on [Jamie Magee's blog](https://jamiemagee.co.uk/blog/maintaining-aur-packages-with-renovate/).
+
+<!-- prettier-ignore -->
+!!! note
+    Jamie Magee helps to maintain Renovate.
+    They obviously like Renovate, and want you to use it.
+
+One big advantage that Arch Linux has over other distributions, apart from being able to say “BTW I use arch”, is the Arch User Repository (AUR).
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:17]:
```suggestion
One big advantage that Arch Linux has over other distributions, apart from being able to say “BTW I use Arch.”, is the Arch User Repository (AUR).
```

Arch is a proper noun, so we should capitalize it. And sentences should end with a full stop.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+- The `matchStrings` is the regex format to extract the version, datasource, and dependency name from the `PKGBUILD`
+- The `extractVersionTemplate` is to handle a “v” in front of any version number that is sometimes added to Git tags
+
+And here’s an extract from the PKGBUILD for the [bicep-bin](https://aur.archlinux.org/packages/bicep-bin) AUR package that I maintain:
+
+```bash
+pkgver=0.15.31 # renovate: datasource=github-tags depName=Azure/bicep
+```
+
+Here I’m configuring Renovate to use the [`github-tags`](https://docs.renovatebot.com/modules/datasource/github-tags/) datasource and to look in the [Azure/bicep GitHub repository](https://github.com/Azure/bicep) for new versions.
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:66]:
```suggestion
Here I’m configuring Renovate to use the [`github-tags`](https://docs.renovatebot.com/modules/datasource/github-tags/) datasource and to look in the [`Azure/bicep` GitHub repository](https://github.com/Azure/bicep) for new versions.
```

Let's monospace the repository name.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+That means it’ll look in the [list of tags in that repository](https://github.com/Azure/bicep/tags) for any new versions.
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:67]:
```suggestion
That means it’ll look in the [list of tags for the `Azure/bicep` repository](https://github.com/Azure/bicep/tags) for any new versions.
```

Use a better link name.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+If Renovate finds any new versions, it’ll automatically update the `PKGBUILD` and open a pull request with the updated version.
+
+So I’ve automated the `PKGBUILD` update, but that’s only half of the work.
+The checksums and `.SRCINFO` must be updated before pushing to the AUR.
+Unfortunately, Renovate can’t do that ([yet](https://github.com/renovatebot/renovate/tree/feat/arch-linux-manager)), but GitHub Actions can!
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:72]: @JamieMagee Is there an issue we can link to? It feels a bit odd to link to a work in progress branch.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+Check out [this pull request for `bicep-bin`](https://github.com/JamieMagee/aur-packages/pull/62) where Renovate opened a pull request, and my GitHub Actions workflow updated the `b2sums` in the `PKGBUILD` and updated the `.SRCINFO`.
+
+But why stop there?
+Let’s talk about publishing.
+
+## Publishing to the AUR
+
+Each AUR package is its own Git repository.
+So to update a package in the AUR, I only need to push a new commit with the updated `PKGBUILD` and `.SRCINFO`.
+Thankfully, [KSXGitHub](https://github.com/KSXGitHub) created the [github-actions-deploy-aur](https://github.com/KSXGitHub/github-actions-deploy-aur) GitHub Action to streamline the whole process.
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:151]:
```suggestion
Thankfully, [KSXGitHub](https://github.com/KSXGitHub) created the [`github-actions-deploy-aur` GitHub Action](https://github.com/KSXGitHub/github-actions-deploy-aur) to streamline the whole process.
```

Monospace for the repository name, and include the GitHub Action part in the link name.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+If I create a new GitHub Actions workflow to publish to the AUR, I can reuse the first two steps from my previous workflow to check out the repository and find the updated package.
+Then all I need to do is to use the [github-actions-deploy-aur](https://github.com/KSXGitHub/github-actions-deploy-aur) GitHub Action:
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:154]:
```suggestion
Then all I need to do is to use the `github-actions-deploy-aur` GitHub Action:
```

We already link to the action on line 151, so we can probably drop the second link.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    pkgname: ${{ env.pkgbuild }}
+    pkgbuild: ${{ env.pkgbuild }}/PKGBUILD
+    commit_username: ${{ secrets.AUR_USERNAME }}
+    commit_email: ${{ secrets.AUR_EMAIL }}
+    ssh_private_key: ${{ secrets.AUR_SSH_PRIVATE_KEY }}
+```
+
+### All together now
+
+If you own any AUR packages and want to automate some of the maintenance burden, check out my [AUR packages template GitHub repository](https://github.com/JamieMagee/aur-packages-template/).
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:170]:
```suggestion
If you own any AUR packages and want to automate some of the maintenance burden, check out [my AUR packages template GitHub repository](https://github.com/JamieMagee/aur-packages-template/).
```

Put the word `my` in the link.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+It contains all of the steps I showed in this blog post.
+And if you want to see how it works in practice, check out my [AUR packages GitHub repository](https://github.com/JamieMagee/aur-packages).
```
Comment (HonkingGoose) [inline docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:172]:
```suggestion
And if you want to see how it works in practice, check out [my AUR packages GitHub repository](https://github.com/JamieMagee/aur-packages).
```

Same thing here.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md (hunk 1 above)
Comment (JamieMagee) [reply docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:34]: I started writing this article a while ago. That probably accounts for the difference.

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md (hunk 5 above)
Comment (JamieMagee) [reply docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:72]: Yep! https://github.com/renovatebot/renovate/issues/16923

**Review by JamieMagee — DISMISSED** [review]

File: docs/usage/user-stories/maintaining-aur-packages-with-renovate.md (hunk 5 above)
Comment (HonkingGoose) [reply docs/usage/user-stories/maintaining-aur-packages-with-renovate.md:72]:
```suggestion
Unfortunately, Renovate can’t do that (yet, see [Renovate issue #16923](https://github.com/renovatebot/renovate/issues/16923)), but GitHub Actions can!
```

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**